src/live_restore/live_restore_server.c

```diff
@@ -155,10 +159,24 @@ __live_restore_worker_run(WT_SESSION_IMPL *session, WT_THREAD *ctx)
     WT_CURSOR *cursor;
     WT_SESSION *wt_session = (WT_SESSION *)session;
     ret = wt_session->open_cursor(wt_session, work_item->uri, NULL, NULL, &cursor);
+    if (ret != 0)
+        __wt_verbose_debug1(session, WT_VERB_LIVE_RESTORE,
+          "Live restore worker: Error opening cursor to %s, ret %d", work_item->uri, ret);
     if (ret == ENOENT) {
         /* Free the work item. */
         __live_restore_free_work_item(session, &work_item);
         return (0);
+    } else if (ret == EBUSY) {
+        /*
+         * Someone else has exclusive access to the table. Add it back to the work queue and try
+         * again later.
+         */
+        __wt_spin_lock(session, &server->queue_lock);
+        TAILQ_INSERT_TAIL(&server->work_queue, work_item, q);
+        __wt_spin_unlock(session, &server->queue_lock);
+        if (server->work_items_remaining < server->threads_working)
```

Comment:
I'd do the read of work items remaining inside the lock in a local var. Ideally we only ever write and read inside the lock eventually.

Reply:
Fixed